src/output-eve.h

```diff
@@ -0,0 +1,60 @@
+/* Copyright (C) 2024 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * \brief EVE logging subsystem
+ *
+ * This file will attempt to the main module for EVE logging
+ * sub-system. Currently most of the API resides in output-json.[ch],
+ * but due to some circular dependencies between EVE, and LogFileCtx,
+ * it made it hard to add EVE filetype modules there until some
+ * include issues are figured out.
+ */
+
+#ifndef __SURICATA_OUTPUT_EVE_H__
```

Comment:
why SURICATA part of the define here? We normally only base on the file name

Reply:
As an include that is part of the library, it should be name spaced as well.  `__OUTPUT_EVE_H__` is probably not that generic, but something simpler like `__OUTPUT_H__`  is more generic with a higher chance of conflicting with a user header.  

LibHTP, libcurl, etc. all do this.

Reply:
should this type of change go into the whole codebase? Perhaps as a separate change/commit?

Reply:
I'm somewhat indifferent here as its not an introducer of merge conflicts like some other changes we have discussed would be.  But would be nice I guess, should we also remove the leading double underscores (and trailing) as they are in reserved namespace?